examples/distributing/uniffi-rust-to-python-library/src/setup.py

```diff
@@ -0,0 +1,229 @@
+from setuptools import setup, Distribution, find_packages
+from setuptools.command.install import install
+
+from distutils.command.build import build as _build
+import os
+import re
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+import wheel.bdist_wheel
+
+sys.dont_write_bytecode = True
+
+
+if sys.version_info < (3, 6):
+    print("the example project requires at least Python 3.6", file=sys.stderr)
+    sys.exit(1)
+
+from pathlib import Path  # noqa
+
+# Path to the directory containing this file
+PYTHON_ROOT = Path(__file__).parent.absolute()
+
+# Relative path to this directory from cwd.
+FROM_TOP = PYTHON_ROOT.relative_to(Path.cwd())
+
+# Path to the root of the git checkout
+SRC_ROOT = PYTHON_ROOT.parents[1]
+
+requirements = [
+    "semver>=2.13.0",
+    "wheel",
+    "semantic-version==2.9.0",
+    "setuptools",
+    "typing_extensions==4.0.1",
+    "pytest==7.1.2",
+    "tox==3.25.1",
+]
+
+buildvariant = "release"
+
+
+class BinaryDistribution(Distribution):
+    def is_pure(self):
+        return False
+
+    def has_ext_modules(self):
+        return True
+
+
+def macos_compat(target):
+    if target.startswith("aarch64-"):
+        return "11.0"
+    return "10.7"
+
+
+# The logic for specifying wheel tags in setuptools/wheel is very complex, hard
```

Comment:
I don't think we need this complexity for this sample - I think it's fine if the target only runs on the Python version it was built with.

Reply:
I agree. This looks like the horrible hack we did over in Glean. And the proper way and IMO what we should recommend is using something like maturin (e.g. see https://github.com/mozilla/glean/pull/2345).
Though for a small example I'm fine with a minimal `setup.py` still.

Reply:
I must watch your PR but it is a draft.

Reply:
@badboy I watched your incomming PR. It seems very much better. @mhammond can I keep the current work as legacy or should I immediately rewrite it please?

Reply:
at this moment I work with maturin.

Reply:
I think @badboy and I are both saying that for this trivial example, there's no need to use maturin. It might make good sense for your project, but for the purposes of this example, we should aim for it being as small as reasonably possible.